net/dns/direct.go

```diff
@@ -371,7 +373,37 @@ func (m *directManager) GetBaseConfig() (OSConfig, error) {
 		fileToRead = backupConf
 	}
 
-	return m.readResolvFile(fileToRead)
+	oscfg, err := m.readResolvFile(fileToRead)
+	if err != nil {
+		return OSConfig{}, err
+	}
+
+	// On some systems, the backup configuration file is actually a
```

Comment:
I think we could simplify this to unconditionally drop tailscale service IPs from configurations we read, no matter what other circumstances are. There is no scenario in which forwarding traffic back to ourselves is correct, that I know of. Did you find a case where this was not true?

That also obviates the need to figure out symlink semantics on all OSes, which is a handy can of worms to keep closed :)

Reply:
I was trying to be maximally-careful here in case we're interoperating with someone else that uses the quad-100 IP... but that probably breaks enough other stuff that it's safe to just drop in all cases, IMO.

Worth dropping unconditionally, or only if we're reading from the backup, do you think?

Reply:
Made it unconditional